Library/Formula/slimerjs.rb

```diff
@@ -25,25 +49,34 @@ def install
     bin.install_symlink libexec/"slimerjs"
   end
 
-  def caveats; <<-EOS.undent
-    You can set the environment variable SLIMERJSLAUNCHER to a installation of
-    Mozilla Firefox (or Mozilla XULRunner) to use this version with SlimerJS
-    instead of the one installed by Homebrew (this is required if built without
-    xulrunner).
-    For a standard Mozilla Firefox installation this would be:
+  def caveats
+    s = ""
 
-      export SLIMERJSLAUNCHER=/Applications/Firefox.app/Contents/MacOS/firefox
+    ffpath = firefox_installation
+    if ffpath.nil?
+      s += <<-EOS.undent
+        You MUST provide an installation of Mozilla Firefox and set
+        the environment variable SLIMERJSLAUNCHER pointing to it, e.g.:
 
-    Note: If you use SlimerJS with an unstable version of Mozilla
-    Firefox/XULRunner (>29.*) you may have to change the [Gecko]MaxVersion in:
-      #{libexec}/application.ini
+        export SLIMERJSLAUNCHER=#{firefox_installation}/Contents/MacOS/firefox
```

Comment:
Recalculation really doesn't matter. Let's just use the function in each location.